Sloth.Core/Services/AggieEnterpriseService.cs

```diff
@@ -44,5 +48,77 @@ public async Task<bool> IsAccountValid(string financialSegmentString, bool valid
 
             return false;
         }
+
+        public async Task<IGlJournalRequestResult> CreateJournal(Source source, Transaction transaction)
+        {
+            if (transaction.Transfers.Count == 0)
+            {
+                throw new ArgumentException("Transaction must have at least one transfer");
+            }
+
+            // all accounting dates must be the same across all transfers, so we can use the first one
+            var accountingDate = transaction.Transfers.First().AccountingDate;
+
+            var lines = new List<GlJournalLineInput>();
+
+            foreach (var transfer in transaction.Transfers)
+            {
+                var line = new GlJournalLineInput
+                {
+                    ExternalSystemIdentifier = transfer.ReferenceId,
+                    ExternalSystemReference = transfer.Id
+                };
+
+                if (transfer.Direction == Transfer.CreditDebit.Credit)
+                {
+                    line.CreditAmount = transfer.Amount;
+                }
+                else
+                {
+                    line.DebitAmount = transfer.Amount;
+                }
+
+                var segmentString = FinancialChartValidation.GetFinancialChartStringType(transfer.FinancialSegmentString);
+
+                if (segmentString == FinancialChartStringType.Gl)
+                {
+                    line.GlSegmentString = transfer.FinancialSegmentString;
+                }
+                else if (segmentString == FinancialChartStringType.Ppm)
+                {
+                    line.PpmSegmentString = transfer.FinancialSegmentString;
+                }
+                else
+                {
+                    throw new ArgumentException("Invalid financial segment string");
+                }
+
+                lines.Add(line);
+            }
+
+            var request = new GlJournalRequestInput
+            {
+                Header = new ActionRequestHeaderInput
+                {
+                    ConsumerTrackingId = transaction.Id,
+                    ConsumerReferenceId = source.Name,
+                    ConsumerNotes = transaction.Description.Substring(0, 240), // TODO: tbd -- link back to sloth?  short desc?  240 chars max.
+                    BoundaryApplicationName = "Sloth",
```

Comment:
Make this Payments, reg, etc?

Reply:
Paymetns, reg, etc are not boundary apps, at least not according to AE team.  We can update to group.Name if needed though later.  It think it would be nice to use individual names somewhere, but not sure AE will want a non-boundary in the boundary app field